ante/gov_ante.go

```diff
@@ -0,0 +1,90 @@
+package ante
+
+import (
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/cosmos/cosmos-sdk/x/authz"
+	govkeeper "github.com/cosmos/cosmos-sdk/x/gov/keeper"
+	govtypes "github.com/cosmos/cosmos-sdk/x/gov/types"
+)
+
+var MiniumInitialDepositRate = sdk.NewDecWithPrec(10, 2)
+
+type GovPreventSpamDecorator struct {
+	govKeeper *govkeeper.Keeper
+	cdc       codec.BinaryCodec
+}
+
+func NewGovPreventSpamDecorator(cdc codec.BinaryCodec, govKeeper *govkeeper.Keeper) GovPreventSpamDecorator {
+	return GovPreventSpamDecorator{
+		govKeeper: govKeeper,
+		cdc:       cdc,
+	}
+}
+
+func (gpsd GovPreventSpamDecorator) AnteHandle(
+	ctx sdk.Context, tx sdk.Tx,
+	simulate bool, next sdk.AnteHandler,
+) (newCtx sdk.Context, err error) {
+	msgs := tx.GetMsgs()
+	if err = gpsd.checkSpamSubmitProposalMsg(ctx, msgs); err != nil {
+		return ctx, err
+	}
+
+	return next(ctx, tx, simulate)
+}
+
+func (gpsd GovPreventSpamDecorator) checkSpamSubmitProposalMsg(ctx sdk.Context, msgs []sdk.Msg) error {
+	validMsg := func(m sdk.Msg) error {
+		if msg, ok := m.(*govtypes.MsgSubmitProposal); ok {
+			// prevent spam gov msg
+			depositParams := gpsd.govKeeper.GetDepositParams(ctx)
+			miniumInitialDeposit := gpsd.calcMiniumInitialDeposit(depositParams.MinDeposit)
+			if msg.InitialDeposit.IsAllLT(miniumInitialDeposit) {
```

Comment:
if deposit `IsAllLT ` than the required deposit, fail.
may use 
if deposit `IsAnyGTE ` than the required deposit, pass. This can do a fast looping, when find one coin `GTE`, return nil ?
both are fine